src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2062 @@
+            similarity scores.
+        logits_per_audio:(`torch.FloatTensor` of shape `(text_batch_size, image_batch_size)`):
+            The scaled dot product scores between `audio_embeds` and `image_embeds`. This represents the audio-image
+            similarity scores.
+        text_embeds(`torch.FloatTensor` of shape `(batch_size, output_dim`):
+            The normalized text embeddings obtained by applying the projection layer to the pooled output of [`ImageBindTextModel`], then applying L2 normalization and logit scaling.
+        image_embeds(`torch.FloatTensor` of shape `(batch_size, output_dim`):
+            The normalized image embeddings obtained by applying the projection layer to the pooled output of [`ImageBindVisionModel`], then applying L2 normalization and logit scaling.
+        audio_embeds(`torch.FloatTensor` of shape `(batch_size, output_dim`):
+            The normalized audio embeddings obtained by applying the projection layer to the pooled output of [`ImageBindAudioModel`], then applying L2 normalization and logit scaling.
+        text_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`ImageBindTextModel`].
+        vision_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`ImageBindVisionModel`].
+        audio_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`ImageBindAudioModel`].
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    logits_per_image: torch.FloatTensor = None
+    logits_per_text: torch.FloatTensor = None
+    logits_per_audio: torch.FloatTensor = None
+    image_embeds: torch.FloatTensor = None
+    text_embeds: torch.FloatTensor = None
+    audio_embeds: torch.FloatTensor = None
+    vision_model_output: BaseModelOutputWithPooling = None
+    text_model_output: BaseModelOutputWithPooling = None
+    audio_model_output: BaseModelOutputWithPooling = None
+
+    def to_tuple(self) -> Tuple[Any]:
+        fields_to_exclude = [
+            "text_model_output",
+            "vision_model_output",
+            "audio_model_output",
+        ]
+        return tuple(self[k] if k not in fields_to_exclude else getattr(self, k).to_tuple() for k in self.keys())
+
+
+class ImageBindGenericPatchEmbedding(nn.Module):
+    """Generic Patch Embedding class that can be used for Vision (image/video), Audio, Depth, Thermal modalities."""
+
+    def __init__(
+        self,
+        config: Union[ImageBindVisionConfig, ImageBindAudioConfig],
+        projection: nn.Module,
+        use_layernorm: bool = False,
+    ):
+        super().__init__()
+
+        if hasattr(config, "image_size"):
+            image_size = config.image_size
+        elif hasattr(config, "num_mel_bins") and hasattr(config, "target_len"):
+            image_size = (config.num_mel_bins, config.target_len)
+        else:
+            raise ValueError("Either `image_size` or `num_mel_bins` and `target_len` must be provided in the config.")
```

Comment:
I think in general here we want to have an abstract `feature_size` for both configs, instead of having if elses!

Reply:
True! It's better that way. Made changes in the recent commits